### PR TITLE
allow KerasTokenizer.numWords to be null

### PR DESCRIPTION
**Pull request by dstine**
## What changes were proposed in this pull request?

This PR allow `KerasTokenizer.numWords` to be `null`. Keras allows `num_words=None`, in fact this is the default.

## How was this patch tested?

I ensured that Keras `TokenizerImportTest` continues to pass, and added an additional test.

## Quick checklist

The following checklist helps ensure your PR is complete:

- [x] Reviewed the [Contributing Guidelines](https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [x] Created tests for any significant new code additions.
- [x] Relevant tests for your changes are passing.
- [x] Ran mvn formatter:format (see [formatter instructions](http://code.revelc.net/formatter-maven-plugin/examples.html#Setting_Source_Files) for targeting your specific files).

### CODE REVIEW

**Comment by dstine** [comment]
For the new test method to pass, a corresponding additional file must be added to the dl4j-test-resources repo. Not sure how you like to handle this kind of inter-dependency. If this change is acceptable, I can make the corresponding pull request over there. For now the new file is a simple copy of the existing `tokenizer.json` with `num_words` set to `null`.

If there are naming standards or patterns to follow (test method, test resource, etc.) please let me know.

File: deeplearning4j/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/preprocessing/text/KerasTokenizer.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ public static KerasTokenizer fromJson(String jsonFileName) throws IOException, I
             throw new InvalidKerasConfigurationException("No configuration found for Keras tokenizer");
 
 
-        int numWords = (int) tokenizerConfig.get("num_words");
+        Integer numWords = (Integer) tokenizerConfig.get("num_words");
```
Comment (dstine) [inline deeplearning4j/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/preprocessing/text/KerasTokenizer.java]: In addition to fixing the NPE, this avoids an unnecessary unbox-then-immediately-rebox. (Same pattern is in use for `documentCount` and could also be changed.)

**Comment by maxpumperla** [comment]
thanks, lgtm. please submit the PR with your test resource first. naming is ok, just make sure it's in the correct relative path.

**Comment by maxpumperla** [comment]
p.s changing other fields like above might make sense as well

**Comment by dstine** [comment]
@maxpumperla 

Thanks for the review. Please see https://github.com/deeplearning4j/dl4j-test-resources/pull/176 with addition of the test resource.

I have another commit with boxing change for `documentCount`. Unless you have a different preferred approach, I can wait to push it until the dl4j-test-resources change is merged so that the new commit will trigger the CI build against the newly-added test resource.

**Comment by maxpumperla** [comment]
merged the resource pr. go ahead with the other change then, please. Let me know once it's ready from your side.

thanks again for this contribution!

**Comment by dstine** [comment]
The CUDA 9.0 and 10.0 builds passed, but 9.2 failed. Seems unlikely to be related to this proposed change. How would you like to proceed?

**Comment by maxpumperla** [comment]
can't be related. I'm happy to merge this

**Comment by dstine** [comment]
Great, that works for me. Thanks!

A couple of questions if you have a moment:
* Any forecast on when a release with this change might be produced?
* I didn’t find a description of the relationship between dl4j-test-resources and this repo. Is that documented and I just couldn’t find it, or is that a gap?

**Comment by agibsonccc** [comment]
We do not comment on release timelines for open source.. 
Dl4j test resources is only for providing datasets and test utilities for unit testing deeplearning4j.
We document that on the repo itself: https://github.com/deeplearning4j/dl4j-test-resources - upon a recent scan of our docs, we need to update this. Thanks for pointing this out cc @ShamsUlAzeem

**Comment by maxpumperla** [comment]
@dstine thanks again. btw, while we do not comment on stable releases, you will find this fix in 1.0.0-SNAPSHOT likely tomorrow.

**Pull request merged**